internal/clients/compliance/client.go

```diff
@@ -41,7 +42,7 @@ func NewClient(conf ComplianceClientConfig) *ComplianceClient {
 }
 
 func (cc *ComplianceClient) request(ctx context.Context, method, url string) (*http.Response, error) {
-	req, err := http.NewRequest(method, url, nil)
+	req, err := http.NewRequestWithContext(ctx, method, url, nil)
```

Comment:
Context must be passed into the request so TracingDoer can pick it up. Also allows us to cancel long-running requests in the future if we need to.